src/kernel/total_sum_accum/cuda.cu

```diff
@@ -20,32 +20,35 @@ namespace nntile::kernel::total_sum_accum
 
 template<typename T>
 static __global__
-void cuda_kernel(Scalar alpha, Index n_labels, Index n_outputs,
+void cuda_kernel(Scalar alpha, Index n_labels, Index n_outputs, Index ignore_index,
         const T *logsumexp, const T *src, const Index *labels, float *val)
 {
     int i = threadIdx.x + blockIdx.x*blockDim.x;
     using Y = typename T::repr_t;
     if(i < n_outputs)
     {
-        __shared__ float block_val;
+        __shared__ Y block_val;
         if(threadIdx.x == 0)
         {
             block_val = 0;
         }
-        float val1 = static_cast<Y>(logsumexp[i]);
-        float val2 = static_cast<Y>(src[labels[i] + i*n_labels]);
+        Y val1 = static_cast<Y>(logsumexp[i]);
+        Y val2 = val1;
+        if (labels[i] != ignore_index)
+            val2 = static_cast<Y>(src[labels[i] + i*n_labels]);
```

Comment:
move `atomicAdd` to the inside of this if statement. if a label is to be ignored, it atomicAdd will just waste GPU cycles to resolve `atomicAdd` data synchronizations.